src/main/java/ladysnake/effective/mixin/WaterFluidMixin.java

```diff
@@ -0,0 +1,36 @@
+package ladysnake.effective.mixin;
+
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.WaterFluid;
+import net.minecraft.particle.ParticleTypes;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import ladysnake.effective.client.world.WaterfallCloudGenerators;
+
+import java.util.Random;
+
+@Mixin(WaterFluid.class)
+abstract class WaterFluidMixin {
+    @Inject(method = "randomDisplayTick", at = @At("RETURN"))
+    private void illuminations$randomDisplayTick(World world, BlockPos pos, FluidState state, Random random, CallbackInfo ci) {
```

Comment:
- Is this actually more performant ? As far as I know, the previous injection only ran upon chunk rebuild, not every frame.
- Is this as reactive as the previous injection ? If I understand correctly, a waterfall could stay undetected for quite a while if it didn't get a random update.

Reply:
> Is this as reactive as the previous injection ?
I tested this and it does update reasonably quickly, yeah.

> Is this actually more performant ?
I didn't notice any measurable difference in performance. The advantage of doing it this way is that it will no longer conflict with Canvas and Sodium and you will no longer need that cursed workaround in #17